### DISCUSSION TITLE
Why base url is deprecated?

### DISCUSSION THREAD

**Discussion by rucciva** [grafana/grafana-infinity-datasource#698] [Q&A]
Hello, first of all, thanks a lot for creating this plugin. it's super awesome.

Just wondering why the Base URL is deprecated? 

I usually need to retrieve data from multiple services with multiple url and each has their own authentication or custom header. In this case, i honestly prefer to create one data source per plugin and have their base url inside the data source settings, while only providing path when using it to create dashboard, thus minimizing duplication of base url.

**Comment by canob** [comment]
Another big problem of deprecating "Base URL": if you want to use a Grafana Datasource Variable with Infinity, for example, to switch between different SaaS products API sources in the same table, with this deprecation, you cannot, because you don´t have the URL configured previously in the datasource.
This is really annoying, and, at the end, it is opposed to the "natural" idea of a Data Source in Grafana, which I think it is that when you "reach" Explore in Grafana, you already have the authentication access to the Data Source configured (that includes headers, query params, and URL in this case), and you can start accessing the data immediately.

**Reply by canob** [reply]
I can add to my comment that the Base URL is not working as expected if you try to use it, by not completing the URL in the Explore Panel:
If you not complete the URL in the Explore Panel, Infinity is going to use your Base URL + the example URL, https://raw.githubusercontent.com/grafana/grafana-infinity-datasource/main/testdata/users.json, + the configured URL Params:
![imagen](https://github.com/grafana/grafana-infinity-datasource/assets/30271927/a8445544-763d-4de6-8ef5-ea6fbe6e85b4)
As a workaround, you can add a SPACE in the URL, and is going to work:
![imagen](https://github.com/grafana/grafana-infinity-datasource/assets/30271927/ed6d7820-c49f-4c23-a742-0da3e3f1f136)

**Reply by canob** [reply]
And another problem, related to the one commented before, but in this case with the deprecation of Global Queries (is it really deprecated, as said in https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/references/global-queries/? Because there is not message about that in the Grafana UI, like the one that you can see in Misc/Base URL, but it is not working right now in 2.8.0 version): for the same scenario mentioned in the previous comment, If you want to switch between different SaaS products API sources in the same table/panel, that different products are going to have different outputs/fields/rows, so you need to "play" with them on "Parsing options & Result fields" of the Global Query to have a different parsing option for each different API product, and in this moment, you cannot, you have the same "Parsing options & Result fields" for every Infinity Datasource that you select using a DataSource Variable for the same panel.
